### PR TITLE
Splitting behaviour on tags

### PR DESCRIPTION
**Pull request by mortenalbertsen**
Two changes are introduced to the vue-input-tag:

- Rather than listening for the keycode pressed, we listen for the key pressed.
This was due to two issues:
1. the fact that keycodes weren't alike across platforms. For instance the space ' ' does not resolve to the same keycode on MacOS (188) and Linux (222) (Windows not tested).
2. the fact that the danish character 'æ' does resolve to the same keycode as ';' (semicolon, keycode 186) does; since we were listening for keycode 186 as a means to separate tags it has the unfortunate effect that danish names such as 'Lærke' and 'Æske' could not be input.

- The ability to provide a custom (regex) pattern used when splitting copy-pasted text. The need for having a custom pattern arose for when entering emails; since a) emails cannot contain spaces and b) more importantly, we wish to allow the user to split input using a space when entering emails.

### CODE REVIEW

File: src/InputTag.vue [hunk 1]
```diff
@@ -32,20 +32,18 @@ export default {
...
       type: Boolean,
       default: false
     },
-    splitExp: {
-      type: String | Object,
-      default: ''
+    regexForSplittingWhenPasting: {
+      type: RegExp,
+      default: function () {
+        return new RegExp(',|;|(?:\\n\\r)|(?:\\n)')
```
Comment (askemottelson) [inline src/InputTag.vue]: why the colon after '?'

File: src/InputTag.vue [hunk 2]
```diff
@@ -61,11 +59,11 @@ export default {
 
   watch: {
     newTag: function (val, old) {
-      if (this.splitExp === '' || this.splitExp === undefined || this.splitExp.test === undefined) {
+      if (this.regexForSplittingWhenPasting === '' || this.regexForSplittingWhenPasting === undefined || this.regexForSplittingWhenPasting.test === undefined) {
```
Comment (askemottelson) [inline src/InputTag.vue]: could you split this in three lines?

File: src/InputTag.vue [hunk 3]
```diff
@@ -96,7 +94,7 @@ export default {
     addNew (e) {
       // Do nothing if the current key code is
       // not within those defined within the addTagOnKeys prop array.
-      if ((e && this.addTagOnKeys.indexOf(e.keyCode) === -1 &&
+      if ((e && this.addTagOnKeys.indexOf(e.key) === -1 &&
```
Comment (askemottelson) [inline src/InputTag.vue]: `x.indexOf(y) === -1` or `x.includes(y)`

**Comment by askemottelson** [comment]
Looks good, I just made a few stylistic comments. About the lookahead: I might be wrong, but just wanted to make sure

File: src/InputTag.vue (hunk 1 above)
Comment (mortenalbertsen) [reply src/InputTag.vue]: 1. So, in general the `?:` syntax denotes a "non-capturing group" - a group that one wishes to be present within the searched string, but one does not subsequently wishes to match on; "I need it to be there, but I don't want to match on it subsequently"
2. There are two issues here: A) the non-capturing group is not needed. B) Using non-capturing groups vs. not using non-capturing groups is inconsistent here. I'll update here, such that we don't use non-capturing groups at all. That should lead to less confusion in the future aswell.

File: src/InputTag.vue [hunk 4]
```diff
@@ -61,11 +59,13 @@ export default {
 
   watch: {
     newTag: function (val, old) {
-      if (this.splitExp === '' || this.splitExp === undefined || this.splitExp.test === undefined) {
+      if (this.regexForSplittingWhenPasting === '' ||
+            this.regexForSplittingWhenPasting === undefined ||
```
Comment (askemottelson) [inline src/InputTag.vue]: something with indentation here :)

File: src/InputTag.vue [hunk 5]
```diff
@@ -32,20 +32,18 @@ export default {
     addTagOnKeys: {
       type: Array,
       default: function () {
-        return [
-          13, // Return
-          188, // Comma ','
-          9 // Tab
```
Comment (askemottelson) [inline src/InputTag.vue]: no tab anymore?

Comment (mortenalbertsen) [reply src/InputTag.vue]: Mistake! Thanks for pointing out! I will fix! 
And add it to the `regexForSplittingWhenPasting `

**Comment by mortenalbertsen** [comment]
- Rather than listening to keycodes, listen for (and react to) the actual character pressed. (Previous issue: (1) Danish character 'æ' resolves to the same keycode as ';' and (2) on Linux, ';' does not resolve to another keycode (222) than the one we were listening for. 
- Explicitate the Regex type of addTagOnKeys
- Add regex to use when attempting to split copy-pasted text.

**Pull request merged**